ext/afform/admin/ang/afGuiEditor/elements/afGuiField.component.js

```diff
@@ -30,14 +36,35 @@
         return !_.isEmpty($scope.meta.searchDisplays);
       };
 
+      this.canBeRange = function() {
+        return this.isSearch() &&
+          !ctrl.getDefn().input_attrs.multiple &&
+          _.includes(['Date', 'Timestamp', 'Integer', 'Float'], ctrl.getDefn().data_type) &&
+          _.includes(['Date', 'Number', 'Select'], $scope.getProp('input_type'));
+      };
+
+      this.canBeMultiple = function() {
+        return this.isSearch() &&
+          !_.includes(['Date', 'Timestamp'], ctrl.getDefn().data_type) &&
+          $scope.getProp('input_type') === 'Select';
+      };
+
+      this.getRangeElements = function(type) {
+        if (!$scope.getProp('search_range') || (type === 'Select' && ctrl.getDefn().input_type === 'Date')) {
+          return singleElement;
+        }
+        return type === 'Date' ? dateRangeElements : rangeElements;
+      };
+
       // Returns the original field definition from metadata
       this.getDefn = function() {
         var defn = afGui.getField(ctrl.container.getFieldEntityType(ctrl.node.name), ctrl.node.name);
-        return defn ||  {
+        defn = defn || {
           label: ts('Untitled'),
-          requred: false,
-          input_attrs: []
+          required: false
         };
+        defn.input_attrs = _.isEmpty(defn.input_attrs) ? {} : defn.input_attrs;
```

Comment:
are we switching from an empty array to an empty object here?

Reply:
Yes the former was a mistake made by my php-centric brain.